### PR TITLE
Add workflows

### PR DESCRIPTION
**Pull request by ederic-oytas**
This pull request adds workflows to test the front-end and back-end for Tabby. For now, these workflows will run on pull request. The front-end workflow only lints for now, and the back-end workflow checks formatting and runs unit tests.

This pull request also configures the linter for the front-end.

### CODE REVIEW

File: server/tabby_server/__main__.py
Comment (David1000734) [inline server/tabby_server/__main__.py:1]: Does the lint from python want no spaces between the operations?
'methods = [...'
has to be replaced with
'methods=[...'

File: .github/workflows/workflow.yml
Comment (David1000734) [inline .github/workflows/workflow.yml:1]: We should consider caching the dependencies in the future to speed up workflow build times

**Review by David1000734 — APPROVED** [review]
Approve

**Comment by ederic-oytas** [comment]
Apologies, it seems that my other commits were also added as well. I updated the README.md in server/ to include the instructions for installing poetry and using poetry install to install all of the dependencies.

**Pull request merged**